### PR TITLE
Setup Profits in Backend

### PR DESCRIPTION
**Pull request by isaacchin**
# Overview

Added Profit entity, ProfitsController, and ProfitsRepository to the backend to support CRUD operations for profits with separate regular user and admin endpoints

# Issues Addressed
#45 

# Details
New endpoints on Swagger, with descriptions:
image

jacoco report:
image

pitest report:
image

Profits still not hooked up to frontend, this PR deals with the backend only

### CODE REVIEW

**Review by vChase42 — DISMISSED** [review]
LGTM

**Review by stychoi — COMMENTED** [review]
LGTM

**Review by stychoi — APPROVED** [review]
LGTM

**Comment by isaacchin** [comment]
Merge conflicts fixed, milkPrice field in Commons entity already exists in main

**Review by vChase42 — DISMISSED** [review]
LGTM

**Review by stychoi — DISMISSED** [review]
LGTM

File: src/main/java/edu/ucsb/cs156/happiercows/entities/Profit.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  private long id;
+
+  @ManyToOne
+  @JoinColumn(name = "user_commons_id")
+  // Allows multiple profit entries in the table to be associated with a single user_commons_id
+
+  private UserCommons userCommons;
+  private long profit;
+  private String time;
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/happiercows/entities/Profit.java:31]: Is is better to use a String time here?  

Or a `LocalDateTime` as in this code from STARTER-team03?

https://github.com/ucsb-cs156-w22/STARTER-team03/blob/a357367a804aa3fc4290a01bd2309a91a2b41c0d/src/main/java/edu/ucsb/cs156/example/entities/UCSBDate.java#L27

**Review by pconrad — CHANGES_REQUESTED** [review]
I wonder whether it's a good idea to use a `String` instead of a `LocalDateTime` for data that is supposed to represent a date/time?

File: src/main/java/edu/ucsb/cs156/happiercows/entities/Profit.java (hunk 1 above)
Comment (isaacchin) [reply src/main/java/edu/ucsb/cs156/happiercows/entities/Profit.java:31]: I was planning to use a library like [Day.js](https://day.js.org/) to parse the String time in the frontend, but if it is better to take care of that sort of thing here then I'll get working on it.

Comment (pconrad) [reply src/main/java/edu/ucsb/cs156/happiercows/entities/Profit.java:31]: So the front end is sort of a separate issue... here what's at issue is "how is the data stored in the database".

Storing it as a LocalDateTime should make it easier to do queries, sorting, etc. based on time.

**Comment by isaacchin** [comment]
Changes requested by Prof. Conrad have been implemented, needs new CR

**Review by vChase42 — APPROVED** [review]
LGTM

**Review by stychoi — APPROVED** [review]
LGTM

**Review by thomashale-ucsb — APPROVED** [review]
Tests pass, and all the functions worked on localhost and heroku. Holy frick that was a ton of backend functions. LGTM.

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**